### DISCUSSION TITLE
AudioGridder in cloud instances

### DISCUSSION THREAD

**Discussion by Deftaudio** [apohl79/audiogridder#1152] [Ideas]
I understand this might be not the original use case for AudioGridder design. I want to bring for consideration the way to run AudioGridder in cloud instances and accessing it remotely. I understand you're sending UDP packets and it seems still doing plain TCP for the discovery, and yes, overall discover mechanism should be changed. Have you ever considered a modification to implement the cloud support?

**Comment by apohl79** [comment]
Latency is the issue. Thats why even wifi is problematic. But you could try it with current implementation as well. It does use mDNS for discovery, but all audio/midi and UI streaming uses tcp. You can connect to your remote server by IP/name. Just add a connection manually.

Of course, there is no auth/encryption etc right now, but you could see if and how it works for you.

**Comment by Deftaudio** [comment]
Thank you for the clarification Andreas.
I understand latency is the issue, no questions here. However, companies like Access Analog exist, where for certain use cases it can be tolerated. Can you please, list all ports that AG is using?

**Reply by apohl79** [reply]
Yeah, it is possible, but the user experience is different. You need more buffering and maybe other optimizations for high latencies.

Ports: 
https://github.com/apohl79/audiogridder/discussions/670#discussioncomment-1088683

**Comment by oldtos** [comment]
A few Discussions below someone was trying this ZeroTier  .. in yet another Discussion someone had success... 
A friend uses sonobus and just remotes in...

**Reply by naokiapri** [reply]
I've tried it from another city with zerotier, cause I'm using audiogridder on windows VM at my unraid server. Rendering was ok, but playback was lagging near to unusable
